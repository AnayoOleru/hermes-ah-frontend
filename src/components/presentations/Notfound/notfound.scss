@import '../../../assets/stylesheets/index.scss'; 

* {
    -webkit-box-sizing: border-box;
            box-sizing: border-box;
  }
  
  body {
    padding: 0;
    margin: 0;
  }
  
  #notfound {
    position: relative;
    height: 70vh;
  }
  
  #notfound .notfound {
    position: absolute;
    left: 50%;
    top: 40%;
    -webkit-transform: translate(-50%, -50%);
        -ms-transform: translate(-50%, -50%);
            transform: translate(-50%, -50%);
  }
  
  .notfound {
    max-width: 410px;
    width: 100%;
    text-align: center;
  }
  
  .notfound .notfound-404 {
    height: 280px;
    position: relative;
    z-index: -1;
  }
  
  .notfound .notfound-404 h1 {
    font-family: $font-stack;
    font-size: 230px;
    margin: 0px;
    font-weight: 900;
    position: absolute;
    left: 50%;
    -webkit-transform: translateX(-50%);
        -ms-transform: translateX(-50%);
            transform: translateX(-50%);
    background: url(https://res.cloudinary.com/mchardex/image/upload/v1557144857/bg.jpg) no-repeat;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    background-size: cover;
    background-position: center;
  }
  
  
  .error-title {
    font-family: $font-stack;
    color: #000;
    font-size: 24px;
    font-weight: 700;
    text-transform: uppercase;
    margin-top: 0;
  }
  
  .error-message {
    font-family: $font-stack;
    color: #000;
    font-size: 14px;
    font-weight: 400;
    margin-bottom: 20px;
    margin-top: 0px;
  }
  
  .home-button{
    font-family: $font-stack;
    font-size: 14px;
    text-decoration: none;
    text-transform: uppercase;
    background: $primary-color;
    display: inline-block;
    padding: 15px 30px;
    border-radius: 40px;
    color: #fff;
    font-weight: 700;
  }
  .home-button:hover{
    background: #ffffff;
    border: 1px solid $primary-color;
    color: rgb(0, 0, 0);
  }
  
  
  @media only screen and (max-width: 767px) {
      .notfound .notfound-404 {
        height: 142px;
      }
      .notfound .notfound-404 h1 {
        font-size: 112px;
      }
  }
