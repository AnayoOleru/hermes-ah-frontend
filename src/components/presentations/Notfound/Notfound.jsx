import React from 'react';
import { Link } from 'react-router-dom';
import './notfound.scss';
import NavBar from '../../shared/NavBar/NavBar';

const Notfound = () => {
  return (
    <React.Fragment>
      <NavBar />
      <div id="notfound">
        <div className="notfound">
          <div className="notfound-404">
            <h1>Oops!</h1>
          </div>
          <h2 className="error-title">404 - Page not found</h2>
          <p className="error-message">
            The page you are looking for might have been removed had its name
            changed or is temporarily unavailable.
          </p>
          <Link className="home-button" to="/">
            Go to homepage
          </Link>
        </div>
      </div>
    </React.Fragment>
  );
};

export default Notfound;
