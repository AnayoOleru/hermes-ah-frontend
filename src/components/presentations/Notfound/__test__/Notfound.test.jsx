import React from 'react';
import { shallow } from 'enzyme';
import Notfound from '../Notfound';

describe('Notfound component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Notfound />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    const wrapper = shallow(<Notfound />);
    expect(wrapper.find('div'));
    expect(wrapper.find('h1'));
    expect(wrapper.find('h2'));
    expect(wrapper.find('p'));
  });
});
