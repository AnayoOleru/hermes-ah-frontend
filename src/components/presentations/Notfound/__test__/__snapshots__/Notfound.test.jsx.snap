// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notfound component should match snapshot 1`] = `ShallowWrapper {}`;
